tests/test_telemetry.py

```diff
@@ -686,9 +686,10 @@ class Test_Metric_Generation_Disabled:
     """Assert that metrics are not reported when metric generation is disabled in telemetry"""
 
     def test_metric_generation_disabled(self):
-        for data in interfaces.library.get_telemetry_data(flatten_message_batches=True):
-            if get_request_type(data) == "generate-metrics":
-                raise Exception("Metric generate event is sent when metric generation is disabled")
+        all_data = list(interfaces.library.get_telemetry_data(flatten_message_batches=True))
+        assert all_data, "No telemetry data to validate on"
```

Comment:
To help no-python people : 

```suggestion
        assert len(all_data) != 0, "No telemetry data to validate on"
```